### PR TITLE
Heart Swap Move Animation

### PR DESCRIPTION
**Pull request by SonikkuA-DatH**
Partially addresses #5192 

https://github.com/user-attachments/assets/7ca91068-8747-486f-a878-2e89b52d7a98

## **Discord contact info**
dat.ha, originally Dat.HA#0697

### CODE REVIEW

**Comment by kittenchilly** [comment]
Looks fine to me overall, but the duplicate skill swap code is kind of annoying. Although it seems to be unavoidable due to the different sprite templates...

**Comment by hedara90** [comment]
Finally had time to look at this properly.
The only comment I have about it is that in doubles, the expanding golden ring effect isn't centered on the 'mon using the move.
pokeemerald-10
While in singles it is centered.
pokeemerald-test-0

File: src/battle_anim_psychic.c [hunk 1]
```diff
@@ -1023,6 +1035,53 @@ void AnimTask_SkillSwap(u8 taskId)
...
+            task->data[10] = 10;
+            task->data[11] = GetBattlerSpriteCoordAttr(gBattleAnimAttacker, BATTLER_COORD_ATTR_LEFT) + 8;
+            task->data[12] = GetBattlerSpriteCoordAttr(gBattleAnimAttacker, BATTLER_COORD_ATTR_BOTTOM) - 8;
+            task->data[13] = GetBattlerSpriteCoordAttr(gBattleAnimTarget, BATTLER_COORD_ATTR_LEFT) + 8;
+            task->data[14] = GetBattlerSpriteCoordAttr(gBattleAnimTarget, BATTLER_COORD_ATTR_BOTTOM) - 8;
+        }
+    }
+    else
+    {
+        if (gBattleAnimArgs[0] == 1)
```
Comment (hedara90) [inline src/battle_anim_psychic.c:1063]: Is the 1 here referring to a specific battle position? If so replace it with the `define`d value of it

**Review by hedara90 — COMMENTED** [review]
For all new functions that have inputs, could you make a comment with that each input is?
e.g. `AnimTask_HeartSwap` takes target as input.

File: src/battle_anim_psychic.c (hunk 1 above)
Comment (SonikkuA-DatH) [reply src/battle_anim_psychic.c]: Yeah it seems to be ANIM_TARGET. Will edit the OG Skill Swap function to update this define as well...

File: src/battle_anim_psychic.c [hunk 2]
```diff
@@ -1023,6 +1035,53 @@ void AnimTask_SkillSwap(u8 taskId)
     task->func = AnimTask_SkillSwap_Step;
 }
```
Comment (hedara90) [inline src/battle_anim_psychic.c:1037]:
```suggestion

// [Explanation of what this task does here]
// Used by: Heart Swap
// arg 0: move target
```
Document what this new anim task does

Comment (SonikkuA-DatH) [reply src/battle_anim_psychic.c:1038]: Alright, should be set. Also defined what the new sprite templates do

File: src/battle_anim_psychic.c [hunk 3]
```diff
@@ -1057,6 +1119,43 @@ static void AnimTask_SkillSwap_Step(u8 taskId)
     }
 }
 
+// Copy of Skill Swap's function to vault the series of orbs between the user and target
+// CreateSprite modified so it uses the pink orbs instead of the blue/green ones
+
```
Comment (hedara90) [inline src/battle_anim_psychic.c:1124]:
```suggestion
```

File: src/battle_anim_psychic.c [hunk 4]
```diff
@@ -1023,6 +1036,55 @@ void AnimTask_SkillSwap(u8 taskId)
     task->func = AnimTask_SkillSwap_Step;
 }
 
+// Copy of Skill Swap's function to get position of the user and target
+// gBattleAnimArgs[0] is the target
```
Comment (hedara90) [inline src/battle_anim_psychic.c:1040]:
```suggestion
// arg 0: move target
```

**Review by hedara90 — APPROVED** [review]

**Pull request merged**